sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -0,0 +1,569 @@
+
+                self.log.error('Error running `%s`: %s', query_method.__name__, error)
+                continue
+
+    def query_master_database(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/20ae63aa7519101496f6b832ec86afbd.html
+        # Only 1 database
+        for master in self.iter_rows(queries.MasterDatabase):
+            tags = ['db:{}'.format(master['db_name']), 'usage:{}'.format(master['usage'])]
+            tags.extend(self._tags)
+
+            master_hostname = master['host']
+            if self._use_hana_hostnames:
+                self._master_hostname = master_hostname
+
+            self.gauge(
+                'uptime',
+                (master['current_time'] - master['start_time']).total_seconds(),
+                tags=tags,
+                hostname=self.get_hana_hostname(master_hostname),
+            )
+
+    def query_database_status(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/dbbdc0d96675470e80801c5ddfb8d348.html
+        for status in self.iter_rows(queries.SystemDatabases):
+            tags = ['db:{}'.format(status['db_name'])]
+            tags.extend(self._tags)
+
+            db_status = self.OK if status['status'].lower() == 'yes' else self.CRITICAL
+            message = status['details'] or None
+            self.service_check(
+                self.SERVICE_CHECK_STATUS, db_status, message=message, tags=tags, hostname=self.get_hana_hostname()
+            )
+
+    def query_backup_status(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/783108ba8b8b4c709959220b4535a010.html
+        for backup in self.iter_rows(queries.GlobalSystemBackupProgress):
+            tags = [
+                'db:{}'.format(backup['db_name']),
+                'service_name:{}'.format(backup['service']),
+                'status:{}'.format(backup['status']),
+            ]
+            tags.extend(self._tags)
+
+            seconds_since_last_backup = (backup['current_time'] - backup['end_time']).total_seconds()
+            self.gauge(
+                'backup.latest', seconds_since_last_backup, tags=tags, hostname=self.get_hana_hostname(backup['host'])
+            )
+
+    def query_licenses(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/1d7e7f52f6574a238c137e17b0840673.html
+        for hana_license in self.iter_rows(queries.GlobalSystemLicenses):
+            tags = ['sid:{}'.format(hana_license['sid']), 'product_name:{}'.format(hana_license['product_name'])]
+            tags.extend(self._tags)
+
+            host = self.get_hana_hostname()
+
+            if hana_license['expiration_date']:
+                expiration = (hana_license['expiration_date'] - hana_license['start_date']).total_seconds()
+            else:
+                expiration = -1
+            self.gauge('license.expiration', expiration, tags=tags, hostname=host)
```

Comment:
I think metric names should be reasonably short with context given by descriptions and the short names. Are you sure you want me to change this?

Reply:
It would be ok, if `license.expiration` mean the expiration date, but here the meaning is the expiration countdown.

`license.expiration` does not really intuitively translate into "time duration until expiration".

Reply:
I mean, since metric values can only be int/float, the expiration date as an arbitrary int/unix timestamp would be useless in practice, so it makes sense that this would be the time until the expiration, similar to 

- https://github.com/DataDog/integrations-core/blob/5287e74e3b44b975dc29e95c09da61c96af35f9f/tls/assets/service_checks.json#L33
- https://github.com/DataDog/integrations-core/blob/5287e74e3b44b975dc29e95c09da61c96af35f9f/vertica/metadata.csv#L9

I can change though, just let me know

Reply:
hum, ok fine, if we already doing that elsewhere. not a big fan though :)